### PR TITLE
Allow saving FieldValues for list Fields if over the limit

### LINKED ISSUE #7542: No values in "a list of all values" filtering widget – metadata.js:228 Missing id
- Your browser and the version: Chrome 66.0.3359.139
- Your operating system: macOS 10.13.4
- Your databases: SQL Server
- Metabase version: 29.0
- Metabase hosting environment: Linux/Ubuntu 12
- Metabase internal database: MySQL

Was super excited to try out the new search widget, but I am unfortunately unable to get it to work. The first time I focus on one of the filters in order to bring up the "Search the list" expanded input field, the following error is being thrown in my console:

```
metadata.js:228 Missing id: {values: ƒ}
copyObjects @ metadata.js:228
(anonymous) @ metadata.js:79
(anonymous) @ index.js:76
(anonymous) @ index.js:36
(anonymous) @ index.js:90
(anonymous) @ index.js:36
(anonymous) @ index.js:86
(anonymous) @ index.js:36
mapStateToProps @ QueryBuilder.jsx:75
mapToPropsProxy @ wrapMapToProps.js:43
handleNewState @ selectorFactory.js:52
handleSubsequentCalls @ selectorFactory.js:69
pureFinalPropsSelector @ selectorFactory.js:74
runComponentSelector @ connectAdvanced.js:26
componentWillReceiveProps @ connectAdvanced.js:150
updateComponent @ ReactCompositeComponent.js:611
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
performUpdateIfNecessary @ ReactCompositeComponent.js:558
performUpdateIfNecessary @ ReactReconciler.js:154
runBatchedUpdates @ ReactUpdates.js:148
perform @ Transaction.js:141
perform @ Transaction.js:141
perform @ ReactUpdates.js:87
flushBatchedUpdates @ ReactUpdates.js:170
closeAll @ Transaction.js:207
perform @ Transaction.js:154
batchedUpdates @ ReactDefaultBatchingStrategy.js:60
enqueueUpdate @ ReactUpdates.js:198
enqueueUpdate @ ReactUpdateQueue.js:22
enqueueSetState @ ReactUpdateQueue.js:216
ReactComponent.setState @ ReactBaseClasses.js:62
onStateChange @ connectAdvanced.js:205
notify @ Subscription.js:26
notifyNestedSubs @ Subscription.js:65
onStateChange @ connectAdvanced.js:202
dispatch @ createStore.js:173
dispatch @ VM329:1
(anonymous) @ middleware.js:22
(anonymous) @ index.js:28
(anonymous) @ store.js:70
(anonymous) @ store.js:17
dispatch @ applyMiddleware.js:35
_callee$ @ redux.js:16
tryCatch @ runtime.js:65
invoke @ runtime.js:303
prototype.(anonymous function) @ runtime.js:117
step @ redux.js:13
(anonymous) @ redux.js:13
Promise.then (async)
step @ redux.js:13
(anonymous) @ redux.js:13
(anonymous) @ redux.js:13
(anonymous) @ redux.js:29
(anonymous) @ store.js:17
(anonymous) @ bindActionCreators.js:3
componentWillMount @ FieldValuesWidget.jsx:86
performInitialMount @ ReactCompositeComponent.js:348
mountComponent @ ReactCompositeComponent.js:255
mountComponent @ ReactReconciler.js:43
performInitialMount @ ReactCompositeComponent.js:368
mountComponent @ ReactCompositeComponent.js:255
mountComponent @ ReactReconciler.js:43
performInitialMount @ ReactCompositeComponent.js:368
mountComponent @ ReactCompositeComponent.js:255
mountComponent @ ReactReconciler.js:43
performInitialMount @ ReactCompositeComponent.js:368
mountComponent @ ReactCompositeComponent.js:255
mountComponent @ ReactReconciler.js:43
mountChildren @ ReactMultiChild.js:234
_createInitialChildren @ ReactDOMComponent.js:701
mountComponent @ ReactDOMComponent.js:520
mountComponent @ ReactReconciler.js:43
performInitialMount @ ReactCompositeComponent.js:368
mountComponent @ ReactCompositeComponent.js:255
mountComponent @ ReactReconciler.js:43
performInitialMount @ ReactCompositeComponent.js:368
mountComponent @ ReactCompositeComponent.js:255
mountComponent @ ReactReconciler.js:43
mountChildren @ ReactMultiChild.js:234
_createInitialChildren @ ReactDOMComponent.js:701
mountComponent @ ReactDOMComponent.js:520
mountComponent @ ReactReconciler.js:43
performInitialMount @ ReactCompositeComponent.js:368
mountComponent @ ReactCompositeComponent.js:255
mountComponent @ ReactReconciler.js:43
performInitialMount @ ReactCompositeComponent.js:368
mountComponent @ ReactCompositeComponent.js:255
mountComponent @ ReactReconciler.js:43
performInitialMount @ ReactCompositeComponent.js:368
mountComponent @ ReactCompositeComponent.js:255
mountComponent @ ReactReconciler.js:43
mountComponentIntoNode @ ReactMount.js:102
perform @ Transaction.js:141
batchedMountComponentIntoNode @ ReactMount.js:124
batchedUpdates @ ReactDefaultBatchingStrategy.js:58
batchedUpdates @ ReactUpdates.js:95
_renderNewRootComponent @ ReactMount.js:317
_renderSubtreeIntoContainer @ ReactMount.js:399
renderSubtreeIntoContainer @ ReactMount.js:340
_renderPopover @ Popover.jsx:276
componentDidMount @ Popover.jsx:91
notifyAll @ CallbackQueue.js:74
close @ ReactReconcileTransaction.js:78
closeAll @ Transaction.js:207
perform @ Transaction.js:154
perform @ Transaction.js:141
perform @ ReactUpdates.js:87
flushBatchedUpdates @ ReactUpdates.js:170
closeAll @ Transaction.js:207
perform @ Transaction.js:154
batchedUpdates @ ReactDefaultBatchingStrategy.js:60
batchedUpdates @ ReactUpdates.js:95
dispatchEvent @ ReactEventListener.js:145
```

I am able to perform the query by manually typing a value into the filter, but the search / autocomplete feature would have really aided the data exploration possibilities.

### PR DESCRIPTION
**Pull request by camsaul**
Includes tests

Fixes #7542 and #7543

### CODE REVIEW

**Comment by camsaul** [comment]
@mazameli can you verify this fix locally?

You will need to re-sync

**Comment by mazameli** [comment]
Sure. Should I also expect this to fix the scenario of upgrading from an earlier version, and having my existing filters using high cardinality fields incorrectly getting set to "a list of values?"

**Comment by mazameli** [comment]
So I set my field with 375 distinct values to "a list of values," and it correctly shows me a list with some, _but not all of the values_ now. If I set it to "search box" then I can search through all of the field's values. My expectation was that I'd be getting a list of all the values in the field in the former situation. I did a manual re-scan of field values after setting it to "a list of values" just to be sure.

**Comment by camsaul** [comment]
@mazameli it wasn't finished yet

try again now everything should be working

**Comment by camsaul** [comment]
Actually @mazameli hold on one more second I need to fix one more thing

**Comment by mazameli** [comment]
Everything appears to be working great now. 🎉

File: docs/users-guide/14-x-rays.md [hunk 1]
```diff
@@ -24,13 +24,13 @@ To quickly make your new dashboard visible to other users, go to the collection
...
 
-![X-ray action in drill-through menu](.images/x-rays/drill-through.png)
+![X-ray action in drill-through menu](./images/x-rays/drill-through.png)
 
 ### X-rays in the Data Reference
 
 You can also create an x-ray by navigating to a table, field, metric, or segment in the [Data Reference](./12-data-model-reference.md). Just click the x-ray link in the left sidebar.
 
-![Data Reference x-ray](.images/x-rays/data-reference.png)
+![Data Reference x-ray](./images/x-rays/data-reference.png)
```
Comment (camsaul) [inline docs/users-guide/14-x-rays.md]: @salsakran these popped up when I switched the targ from 29.0 to 29.1

Are these changes ok?

Comment (salsakran) [reply docs/users-guide/14-x-rays.md]: Hmmm. I branched 0.29.1 off 0.29.0 and those went in when we found broken links during the release checks. 

The ones with the slash are definitely correct.

**Pull request merged**